src/main/java/uk/gov/mca/beacons/service/registrations/BeaconsController.java

```diff
@@ -28,8 +30,16 @@ public BeaconsSearchResult findAll() {
     return results;
   }
 
-  @GetMapping(value = "/{hexId}")
-  public Beacon findByHexId(@PathVariable("hexId") String hexId) {
-    return new Beacon();
+  @GetMapping(value = "/{uuid}")
+  public WrapperDTO<BeaconDTO> findByUuid(
+    @PathVariable("uuid") String uuidString
```

Comment:
Spring should deserialise the argument into a type `UUID` for you so you should be able to do:

```java
    @PathVariable("uuid") UUID id
```